packages/cli-platform-android/src/commands/buildAndroid/startServerInNewWindow.ts

```diff
@@ -6,44 +6,55 @@ import {logger} from '@react-native-community/cli-tools';
 export function startServerInNewWindow(
   port: number,
   terminal: string,
+  projectRoot: string,
   reactNativePath: string,
 ) {
   /**
    * Set up OS-specific filenames and commands
    */
   const isWindows = /^win/.test(process.platform);
-  const scriptFile = isWindows
-    ? 'launchPackager.bat'
-    : 'launchPackager.command';
+  const scriptFile = isWindows ? 'launchPackager.bat' : 'launchPackager.sh';
   const packagerEnvFilename = isWindows ? '.packager.bat' : '.packager.env';
-  const portExportContent = isWindows
-    ? `set RCT_METRO_PORT=${port}`
-    : `export RCT_METRO_PORT=${port}`;
+  const packagerEnvFileExportContent = isWindows
+    ? `set RCT_METRO_PORT=${port}\nset PROJECT_ROOT=${projectRoot}\nset REACT_NATIVE_PATH=${reactNativePath}`
+    : `export RCT_METRO_PORT=${port}\nexport PROJECT_ROOT=${projectRoot}\nexport REACT_NATIVE_PATH=${reactNativePath}`;
+  const nodeModulesPath = path.join(projectRoot, 'node_modules', '.bin');
+  const cliPluginMetroPath = path.dirname(
+    require.resolve('@react-native-community/cli-plugin-metro/package.json'),
+  );
 
   /**
-   * Set up the `.packager.(env|bat)` file to ensure the packager starts on the right port.
+   * Set up the `.packager.(env|bat)` file to ensure the packager starts on the right port and in right directory.
    */
-  const launchPackagerScript = path.join(
-    reactNativePath,
-    `scripts/${scriptFile}`,
-  );
+  const packagerEnvFile = path.join(nodeModulesPath, `${packagerEnvFilename}`);
 
   /**
-   * Set up the `launchpackager.(command|bat)` file.
+   * Set up the `launchPackager.(command|bat)` file.
    * It lives next to `.packager.(bat|env)`
    */
-  const scriptsDir = path.dirname(launchPackagerScript);
-  const packagerEnvFile = path.join(scriptsDir, packagerEnvFilename);
-  const procConfig: execa.SyncOptions = {cwd: scriptsDir};
+  const launchPackagerScript = path.join(nodeModulesPath, scriptFile);
+  const procConfig: execa.SyncOptions = {cwd: path.dirname(packagerEnvFile)};
 
   /**
    * Ensure we overwrite file by passing the `w` flag
    */
-  fs.writeFileSync(packagerEnvFile, portExportContent, {
+  fs.writeFileSync(packagerEnvFile, packagerEnvFileExportContent, {
     encoding: 'utf8',
     flag: 'w',
   });
 
+  /**
+   * Copy files into `node_modules/.bin`.
+   */
+  fs.copyFileSync(
+    path.join(cliPluginMetroPath, 'launchPackager.bat'),
+    path.join(nodeModulesPath, 'launchPackager.bat'),
+  );
+  fs.copyFileSync(
+    path.join(cliPluginMetroPath, 'launchPackager.sh'),
+    path.join(nodeModulesPath, 'launchPackager.sh'),
+  );
```

Comment:
let's add some exception handling here, letting users know what happened and what they can do (e.g. retry)

Reply:
done! ✅